# Ejercico de casa: Regresión Múltiple

In [1]:
# Instala dependencias
install.packages(c("rstatix", "QuantPsyc", "olsrr"))
library(tidyverse)
library(rstatix)
library(QuantPsyc)
library(olsrr)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependencies ‘numDeriv’, ‘SparseM’, ‘MatrixModels’, ‘minqa’, ‘nloptr’, ‘Rcpp’, ‘RcppEigen’, ‘carData’, ‘abind’, ‘pbkrtest’, ‘quantreg’, ‘lme4’, ‘corrplot’, ‘car’


── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors

Attaching package: ‘rstatix’


The following object is masked from ‘package:stats’:

    filter




In [2]:
df = haven::read_sav("https://github.com/renatoparedes/EstadisticaYPsicologiaMatematica/raw/main/INEE/BaseEjerciciosRegresionMultiple.sav")

df

Código Mot_Log Autoconcepto Rend_Acad
1  001     8      8            18       
2  002    10      8            19       
3  003     8      6            18       
4  004     8      7            19       
5  005     7      8            18       
6  006     6      8            15       
7  007     9      6            19       
8  008     6      8            18       
9  009     6      8            14       
10 010     8      8            18       
11 011     5      5            18       
12 012     8      8            19       
13 013     5      6            17       
14 014     7      7            18       
15 015     7      8            17       
16 016     9      7            17       
17 017     7      9            18       
18 018     7      9            20       
19 019     6      7            17       
20 020     7      7            18       
21 021     7      5            17       
22 022     7      7            16       
23 023     5      7            16       
24 024     5      5            16       
25 025     8      9            17       
26 026     5      5            16       
27 027     7      8            16       
28 028     5      7            17       
29 029     7      5            16       
30 030     5      5            16       
31 031     5      6            16       
32 032     6      5            16       
33 033     5      4            16       
34 034     4      5            16       
35 035     4      6            13       
36 036     5      5            15       
37 037     5      5            15       
38 038     7      7            15       
39 039     5      5            15       
40 040     6      5            15       
41 041     5      7            15       
42 042     4      5            15       
43 043     5      6            15       
44 044     5      5            15       
45 045     6      7            15       
46 046     7      8            18       
47 047     7      9            19       
48 048     8      8            19       
49 049     5      6            14       
50 050     3      5            14       
51 051     5      5            13       
52 052     4      7            14       
53 053     8      5            13       
54 054     4      7            13       
55 055     7      5            13       
56 056     4      6            13       
57 057     4      5            13       
58 058     6      7            14       
59 059     4      5            13       
60 060     7      8            15

## Ejercicio 1

El estudio sobre los factores presentes en el rendimiento académico en estudiantes universitarios encontró que tanto la motivación de logro como el autoconcepto son variables predictoras significativas del rendimiento de las y los participantes. Ahora, se quiere conocer cuál es la contribución conjunta de ambos factores a la variabilidad del rendimiento académico y cuál de ellos es la variable más importante en su estimación.

a) Establece la hipótesis de trabajo (HT).

b) Señala las hipótesis estadísticas (H0 y H1) que correspondan a la HT.

c) Plantea la ecuación del modelo para la HT.

d) Evalúa los supuestos necesarios para continuar con el análisis de regresión.
- Linealidad
- Normalidad de residuos
- Multicolinealidad

e) Evalúa el ajuste del modelo planteado, interpretando el valor de R2 en términos de variabilidad explicada del rendimiento académico y los criterios de Cohen.

f) Indica si se cumplen las hipótesis y brinda tu conclusión.


In [3]:
df %>% shapiro_test(Rend_Acad, Mot_Log, Autoconcepto)

variable     statistic p           
1 Autoconcepto 0.8853151 3.997856e-05
2 Mot_Log      0.9408834 5.932201e-03
3 Rend_Acad    0.9427623 7.211886e-03

A partir de la prueba de normalidad de Shapiro-Wilk, se observa que tanto el autoconcepto, $SW(60) = .885, p < .001$, la motivación de logro, $SW(60) = .941, p = .005$, y en rendimiento académico,$SW(60) = .943, p = .007$, no muestran una distribución normal.

In [6]:
df %>% cor_test(vars = c("Rend_Acad"), vars2 = c("Mot_Log", "Autoconcepto"), method = "spearman")

var1      var2         cor  statistic p        method  
1 Rend_Acad Mot_Log      0.65 12694.15  2.29e-08 Spearman
2 Rend_Acad Autoconcepto 0.50 18031.91  4.95e-05 Spearman

A partir de la prueba de correlación de Spearman, se observa que existe una relación estadísticamente significativa, positiva y grande entre motivación de logro, $r(58) = .65, p < .001$ y autoconcepto, $r(58) = .50, p < .001$, con el rendimiento académico.

In [8]:
df %>%
  lm(Rend_Acad ~ Mot_Log + Autoconcepto, data = .) -> model

model$residuals %>%
  shapiro_test()

variable statistic p.value  
1 .        0.9703446 0.1517005

A partir de la prueba de normalidad de Shapiro-Wilk, se observa que existe una distribución normal de los residuos estandarizados, $SW(60) = .970, p = .152$.

In [13]:
ols_vif_tol(model)

Variables    Tolerance VIF    
1 Mot_Log      0.7210999 1.38677
2 Autoconcepto 0.7210999 1.38677


Motivación de Logro
*   VIF = 1.387 < 3
*   TOL = .721 > .2

Autoconcepto
*   VIF = 1.387 < 3
*   TOL = .721 > .2


Dado que la tolerancia es mayor a .2 y el VIF es < .3 en cada variable, se asume que estas son independientes.

In [14]:
model %>% broom::glance()

r.squared adj.r.squared sigma    statistic p.value     df logLik    AIC     
1 0.4572429 0.4381988     1.450154 24.00968  2.73059e-08 2  -105.8977 219.7954
  BIC      deviance df.residual nobs
1 228.1727 119.8679 57          60

A partir del ANOVA, se observa que el modelo es adecuado para ser interpretado, $F(2, 57) = 24.010, p < .001$. El coeficiente de determinación es grande, según el criterio de Cohen (1988), y explica el 43.8% de la variabilidad del rendimiento académico, $R^2 = .438$.

In [15]:
model %>% broom::tidy()

term         estimate  std.error statistic p.value     
1 (Intercept)  9.8684396 0.9704715 10.168706 2.029447e-14
2 Mot_Log      0.6529186 0.1450644  4.500889 3.389025e-05
3 Autoconcepto 0.3399444 0.1620895  2.097264 4.041711e-02

A partir de la regresión lineal múltiple, se observa que existe un modelo adecuado, con un tamaño grande, y que explica el 43.6% de la variabilida del rendimiento, $R^2 = .438, F(2, 57) = 24.010, p < .001$. Específicamente, tanto la motivación de logro, $\beta = .653, EE_\beta = .15, t(57) = 4.50, p < .001$, como el autoconcepto, $\beta = .340, EE_\beta = .16, t(57) = 2.10, p = .04$, son predictores estadísticamente significativos y positivos del rendimiento académico.

In [16]:
lm.beta(model) %>% enframe()

name         value    
1 Mot_Log      0.5172088
2 Autoconcepto 0.2410020

La variable mas importante del modelo es la motivación de logro, $B = .517$.

$R = 9.868 + .653(M) + .340(A) \pm 1.45$